packages/core/useStorageAsync/queue.ts

```diff
@@ -0,0 +1,32 @@
+export class TaskQueue {
+  private readonly queue: ((() => Promise<void>) | null)[]
+  private readonly emptyCallback: () => void
+
+  constructor(callback?: () => void) {
+    this.queue = [null, null]
+    this.emptyCallback = callback ?? (() => {})
+  }
+
+  enqueue(task: () => Promise<void>) {
+    this.queue[1] = task
```

Comment:
Yikes!

Basically we just want to pause the watcher while the async read method is running, we have composables for that which could be used instead of this very shaky TaskQueue